File: parent-337/pinyougou-web-manager/src/main/webapp/js/service/specificationservice.js
app.service("specificationService", function ($http) {
    // 分页+条件查询规格
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../specification/search.do?pageNum=' + pageNum + '&pageSize=' + pageSize, searchEntity);
    };

    // 添加规格的保存
    this.add = function (entity) {
        return $http.post('../specification/add.do', entity);
    }

    // 修改规格的保存
    this.update = function (entity) {
        return $http.post('../specification/update.do', entity);
    }

    // 修改规格之数据回显
    this.findOne = function (id) {
        return $http.get('../specification/findOne.do?id=' + id);
    }

    // 删除规格
    this.del = function (ids) {
        return $http.get('../specification/del.do?ids=' + ids);
    }

    // 查询规格id和规格的名字,并存入map集合中
    this.selectOptionList = function () {
        return $http.get('../specification/selectOptionList.do');
    }
});

File: parent-337/pinyougou-interface/src/main/java/cn/ithcast/core/service/SpecificationService.java
package cn.ithcast.core.service;

import cn.itcast.core.pojo.specification.Specification;
import cn.itcast.core.pojogroup.SpecificationVo;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface SpecificationService {

    /**
     * 分页+条件查询规格
     *
     * @param pageNum
     * @param pageSize
     * @param specification
     * @return
     */
    PageResult search(Integer pageNum, Integer pageSize, Specification specification);

    /**
     * 添加规格及规格选项
     */
    public void add(SpecificationVo specificationVo);

    /**
     * 修改规格之数据回显
     *
     * @param id
     * @return
     */
    SpecificationVo findOne(Long id);

    /**
     * 修改规格之修改规格
     *
     * @param specificationVo
     */
    void update(SpecificationVo specificationVo);

    /**
     * 删除规格
     *
     * @param ids
     */
    void del(Long[] ids);

    /**
     * 查询规格id和规格的名字,并存入map集合中
     */
    List<Map> selectOptionList();
}
